### PR TITLE
Tweaks

### PR DESCRIPTION
**Pull request by willrogers**
@nickbattam @mfurseman

### CODE REVIEW

File: opimodel/widgets.py [hunk 1]
```diff
@@ -232,6 +232,8 @@ class GroupingContainer(Widget):
     def __init__(self, x, y, width, height):
         super(GroupingContainer, self).__init__(
             GroupingContainer.TYPE_ID, x, y, width, height)
+        self.lock_children = True
```
Comment (nickbattam) [inline opimodel/widgets.py]: why lock_children be default; this is an annoying setting if you're trying to manually update panels and is invisible to anyone running them.

File: opimodel/widgets.py [hunk 2]
```diff
@@ -254,30 +258,31 @@ def __init__(self, x, y, width, height, text):
...
         self.square_button = True
         self.show_boolean_label = True
         self.show_led = False
+        self.push_action_index = 0
+        self.released_action_index = 1
 
     def add_push_action(self, action):
-        self.actions.append(action)
+        self.actions.add_action(action)
         self.push_action_index = len(self.actions) - 1
```
Comment (nickbattam) [inline opimodel/widgets.py:281]: This is the index of the just added (i.e. last) action. The test asserts that the action should be hooked to the first action.

File: opimodel/widgets.py [hunk 3]
```diff
@@ -254,30 +258,31 @@ def __init__(self, x, y, width, height, text):
...
 
     def add_release_action(self, action):
-        self.actions.append(action)
-        self.release_action_index = len(self.actions) - 1
+        self.actions.add_action(action)
+        self.released_action_index = len(self.actions) - 1
```
Comment (nickbattam) [inline opimodel/widgets.py:286]: Same as for push_action

File: opimodel/widgets.py [hunk 4]
```diff
@@ -221,6 +222,7 @@ def __init__(self, x, y, width, height, pv):
             TextInput.TYPE_ID, x, y, width, height)
 
         self.pv_name = pv
+        self.horizontal_alignment = HAlign.CENTER
```
Comment (mfurseman) [inline opimodel/widgets.py:225]: Is this necessarily better than whatever CS-Studio inserts as the default alignment?

Comment (willrogers) [reply opimodel/widgets.py:225]: Yes.

Well it's closer to our schema, which is the point really.

File: opimodel/widgets.py (hunk 2 above)
Comment (willrogers) [reply opimodel/widgets.py:281]: I think this is correct, so I think this is a comment on the structure of the test (since the test passes).

Comment (nickbattam) [reply opimodel/widgets.py:281]: OK, I was confused.

Looking deeper at a BoolButton added in the OPIEditor it has:
    `<actions hook="false" hook_all="false">`
As ToggleButton extends ActionWidget it picks up the default args of `hook_first=True, hook_all=False` and doesn't set `hook` at all.

File: opimodel/widgets.py (hunk 4 above)
Comment (nickbattam) [reply opimodel/widgets.py:225]: Need to add a test that TextUpdate and TextInput widgets are correctly aligned

**Comment by willrogers** [comment]
Can we merge this?

**Pull request merged**